src/Analyzers/CSharp/Tests/NamingStyles/NamingStylesTests.cs

```diff
@@ -1382,5 +1382,31 @@ class C : I
     int [|global::I.X|] => 0;
 }", new TestParameters(options: s_options.PropertyNamesArePascalCase));
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.NamingStyle)]
+        public async Task TestAsyncEntryPoint()
+        {
+            await TestMissingInRegularAndScriptAsync(@"
+using System.Threading.Tasks;
+
+class C
+{
+    static async Task [|Main|]()
```

Comment:
This is different than the linked issue, but it fails currently. I'll work on getting that fix.